explorer/ast/declaration.h

```diff
@@ -295,12 +293,24 @@ class ClassDeclaration : public Declaration {
 
   auto value_category() const -> ValueCategory { return ValueCategory::Let; }
 
+  auto base_expr() const -> std::optional<Nonnull<Expression*>> {
+    return base_expr_;
+  }
   auto base() const -> std::optional<Nonnull<const ClassDeclaration*>> {
     return base_;
   }
   void set_base(Nonnull<const ClassDeclaration*> base_decl) {
     base_ = base_decl;
   }
```

Comment:
Can we remove this? It doesn't seem like it can be correct to use this when the base can be something like `X(i32)`, and so any callers should be using `base_type` instead. Are there any callers of `base` left?

Reply:
Good point, removed and swapped `base()` calls for `base_type()->declaration()`;